apps/publisher/src/app.tsx

```diff
@@ -1,56 +1,59 @@
+  WrapItem,
 } from '@chakra-ui/react';
-import {
-  IconCameraOff,
-  IconCameraOn,
-  IconClose,
-  IconInfo,
-  IconMicrophoneOff,
-  IconMicrophoneOn,
-  IconPresent,
-  IconProfile,
-} from '@millicast-react/dolbyio-icons';
-import './styles/font.css';
-import usePublisher from '@millicast-react/use-publisher';
-import type { SourceState } from '@millicast-react/use-publisher';
-import useMediaDevices from '@millicast-react/use-media-devices';
+import { StreamStats, VideoCodec } from '@millicast/sdk';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import ActionBar from '@millicast-react/action-bar';
+import DeviceSelection from '@millicast-react/device-selection';
+import { IconAddCamera, IconPresent, IconProfile } from '@millicast-react/dolbyio-icons';
+import InfoLabel from '@millicast-react/info-label';
+import LiveIndicator from '@millicast-react/live-indicator';
 import ParticipantCount from '@millicast-react/participant-count';
-import ShareLinkButton from '@millicast-react/share-link-button';
 import PopupMenu from '@millicast-react/popup-menu';
-import LiveIndicator from '@millicast-react/live-indicator';
+import ShareLinkButton from '@millicast-react/share-link-button';
 import Timer from '@millicast-react/timer';
-import ActionBar from '@millicast-react/action-bar';
-import ControlBar from '@millicast-react/control-bar';
-import InfoLabel from '@millicast-react/info-label';
+import useMultiMediaDevices, { Resolution, StreamId, StreamTypes } from '@millicast-react/use-multi-media-devices';
 import useNotification from '@millicast-react/use-notification';
-import type { StreamStats, VideoCodec } from '@millicast/sdk';
+import usePublisher, { SourceState } from '@millicast-react/use-publisher';
 
 import PublisherVideoView from './components/publisher-video-view';
-import useCameraCapabilities, { Resolution } from './hooks/use-camera-capabilities';
+import './styles/font.css';
+
+const MAX_SOURCES = 4;
 
-//TODO: Support more than 2 sources
-const MainSourceId = 'Main';
-const DisplayShareSourceId = 'DisplayShare';
 type PublisherState = SourceState;
 
 const date = new Date();
 
 function App() {
+  const {
+    isOpen: isDeviceSelectionOpen,
+    onOpen: handleOpenDeviceSelection,
+    onClose: handleCloseDeviceSelection,
+  } = useDisclosure();
+
+  const [bitrates, setBitrates] = useState<Map<StreamId, number>>(new Map());
+  const [codecs, setCodecs] = useState<Map<StreamId, VideoCodec>>(new Map());
+  const [isSimulcastEnabled, setIsSimulcastEnabled] = useState(true);
+  const [labels, setLabels] = useState<Map<StreamId, string>>(new Map());
+  const [newCamera, setNewCamera] = useState<InputDeviceInfo | undefined>(undefined);
+  const [newMicrophone, setNewMicrophone] = useState<InputDeviceInfo | undefined>(undefined);
+  const [publisherStates, setPublisherStates] = useState<Map<StreamId, PublisherState>>(new Map());
+  const [statistics, setStatistics] = useState<Map<StreamId, StreamStats>>(new Map());
```

Comment:
I extended off the existing patterns for `bitrate`, `codec`, `publisherState` by changing them to `Map` in order to support multiple sources.

Reply:
I thought they should stay in the `sources` which are returned by the `usePublisher` hook, so the application doesn't need to maintain these, it only calls the API from `usePublisher` hook to manipulate them.

Reply:
TBD: refactor `usePublisher` hook to store all stream/source states
https://github.com/dolbyio-samples/rts-app-react-publisher-viewer/issues/241